custom_libs/subliminal_patch/providers/jimaku.py

```diff
@@ -0,0 +1,421 @@
+from __future__ import absolute_import
+
+import io
+import logging
+import re
+import time
+import traceback
+import urllib.parse
+import rarfile
+import zipfile
+import requests
+import xml.etree.ElementTree as etree
+
+from requests import Session
+from subliminal import __short_version__
+from subliminal.exceptions import ConfigurationError
+from subliminal.utils import sanitize
+from subliminal.video import Episode, Movie
+from subliminal_patch.providers import Provider
+from subliminal_patch.subtitle import Subtitle
+from subliminal_patch.exceptions import TooManyRequests
+from subliminal_patch.providers.utils import get_subtitle_from_archive
+from functools import cache
+from subzero.language import Language
+
+logger = logging.getLogger(__name__)
+
+class JimakuSubtitle(Subtitle):
+    '''Jimaku Subtitle.'''
+    provider_name = 'jimaku'
+    
+    hash_verifiable = False
+
+    def __init__(self, video, subtitle_id, subtitle_url, release_info, anilist_id=None):
+        # Override param 'language' as it could only ever be "ja"
+        super(JimakuSubtitle, self).__init__(Language("jpn"))
+        
+        self.video = video
+        self.subtitle_id = subtitle_id
+        self.subtitle_url = subtitle_url
+        self.release_info = release_info
+        self.anilist_id = anilist_id
+        
+    @property
+    def id(self):
+        return self.subtitle_id
+
+    def get_matches(self, video):
+        matches = set()
+        
+        if isinstance(video, Episode):
+            # series name
+            if sanitize(video.series) and sanitize(self.video.series) in (
+                    sanitize(name) for name in [video.series] + video.alternative_series):
+                matches.add('series')
+            
+            # season
+            if video.season and self.video.season is None or video.season and video.season == self.video.season:
+                matches.add('season')            
+
+            # year
+            if video.original_series and self.video.year is None or video.year and video.year == self.video.year:
+                matches.add('year')
+                
+            # episode
+            if video.episode == self.video.episode:
+                matches.add('episode')
+        else:
+            raise ValueError(f"Unhandled instance of argument 'video': {type(video)}")
+
+        return matches
+
+class JimakuProvider(Provider):
+    '''Jimaku Provider.'''
+    video_types = (Episode, Movie)
+    api_url = 'https://jimaku.cc/api'
+    
+    api_ratelimit_max_delay = 5 # In seconds
```

Comment:
This is probably a better name instead of comment?

```suggestion
    api_ratelimit_max_delay_seconds = 5 # In seconds
```

Reply:
Implemented suggestion.